docs/languages/csharp-properties.md

```diff
@@ -0,0 +1,253 @@
+fields. A field defines a storage location:
+
+```
+public class Person
+{
+	public string FirstName;
+	// elided
+}
+```
+
+A property definition contains declarations for a `get` and
+`set` accessor that retrieves and assigns the value of that
+property:
+
+```
+public class Person
+{
+	public string FirstName
+	{
+		get;
+		set;
+	}
+	// elided
+}
+```
+
+The syntax shown above is the *auto property* syntax. The compiler
+generates the storage location for the field that backs up the 
+property. The compiler also implements the body of the `get` and `set` accessors.
+You can also define the storage yourself, as shown below:
+
+```
+public class Person
+{
+	public string FirstName
+	{
+		get { return firstName; }
+		set { firstName = value; }
+	}
+	private string firstName;
+	// elided
+}
+```
+ 
+The property definition shown above is a read-write property. Notice
+the keyword `value` in the set accessor. The `set` accessor always has
+a single parameter named `value`. The `get` accessor must return a value
+that is convertable to the type of the property (`string` in this example).
+ 
+That's the basics of the syntax. There are many different variations that support
+a variety of different design idioms. Let's explore those, and learn the syntax
+options for each. 
+
+# Scenarios
+
+The examples above showed one of the simplest cases of property definition:
+a read-write property with no validation. By writing the code you want in the
+`get` and `set` accessors, you can create many different scenarios.  
+
+## Validation
+
+You can write code in the `set` accessor to ensure that the values represented
+by a property are always valid. For example, suppose one rule for the `Person`
+class is that the name cannot be blank, or whitespace. You would write that as
+follows:
+
+```
+public class Person
+{
+    public string FirstName
+    {
+        get { return firstName; }
+        set
+        {
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentException("First name must not be blank");
+            firstName = value;
+        }
+    }
+    private string firstName;
+    // elided
+}
+```
+
+The example above enforces the rule that the first name must not be blank,
+or whitespace. If a developer writes
+```
+foo.FirstName = "";
+```
+That assignment throws an `ArgumentException`. Because a property set accessor
+must have a void return type, you report errors in the set accessor by throwing an exception.
+
+That is a simple case of validation. You can extend this same syntax to anything needed
+in your scenario: You can check the relationships between different properties, or validate
+against any external conditions. Any valid C# statements are valid in a property accessor.
+
+## INotifyPropertyChanged
+
+Another common scenario where you need to write code in a property accessor is to
+support the `INotifyPropertyChanged` interface used in XAML based data binding.
```

Comment:
Instead of mentioning XAML here, should we say something more generic like "used to notify binding clients that a property value has changed". Do we also need to explain what data binding is before explaining how the interface works?